core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -329,11 +331,51 @@ private void tryStart() {
             }
 
             // Wait until the process within the container has become ready for use (e.g. listening on network, log message emitted, etc).
-            waitUntilContainerStarted();
+            try {
+                waitUntilContainerStarted();
+            } catch (Exception e) {
+                logger().debug("Wait strategy threw an exception", e);
+                InspectContainerResponse inspectContainerResponse = null;
+                try {
+                    inspectContainerResponse = dockerClient.inspectContainerCmd(containerId).exec();
+                } catch (NotFoundException notFoundException) {
+                    logger().debug("Container {} not found", containerId, notFoundException);
+                }
+
+                if (inspectContainerResponse == null) {
+                    throw new IllegalStateException("Container is removed");
+                }
+
+                InspectContainerResponse.ContainerState state = inspectContainerResponse.getState();
+                if (Boolean.TRUE.equals(state.getDead())) {
+                    throw new IllegalStateException("Container is dead");
+                }
+
+                if (Boolean.TRUE.equals(state.getOOMKilled())) {
+                    throw new IllegalStateException("Container crashed with out-of-memory");
```

Comment:
Super trivial, but could we include the string `OOMKilled` in the message? i.e.

```suggestion
                    throw new IllegalStateException("Container crashed with out-of-memory (OOMKilled)");
```

`OOMKilled` as a term has become fairly standard, with a specific meaning, so I think it's better to keep the specificity of the error as we propagate it.